src/main/scala/org/ergoplatform/mining/ErgoMiner.scala

```diff
@@ -220,34 +223,56 @@ class ErgoMiner(ergoSettings: ErgoSettings,
   // helper method to update cached candidate block and corresponding message for external miners
   private def updateCandidate(candidate: CandidateBlock,
                               pk: ProveDlog,
-                              txsToInclude: Seq[ErgoTransaction]): WorkMessage = {
+                              txsToInclude: Seq[ErgoTransaction]): CandidateCache = {
     val ext = powScheme.deriveExternalCandidate(candidate, pk, txsToInclude.map(_.id))
     log.info(s"New candidate with msg ${Base16.encode(ext.msg)} generated")
     log.debug(s"Got candidate block at height ${ErgoHistory.heightOf(candidate.parentOpt) + 1}" +
       s" with ${candidate.transactions.size} transactions")
-    candidateOpt = Some(CandidateCache(candidate, ext, txsToInclude))
+    val candCache = CandidateCache(candidate, ext, txsToInclude)
+    candidateOpt = Some(candCache)
     if (!externalMinerMode) miningThreads.foreach(_ ! candidate)
-    ext
+    candCache
```

Comment:
Please comment on the logic of sending the same candidate to all the threads in parallel.